hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py

```diff
@@ -1,33 +1,96 @@
+        for action_label in action_labels:
+            perform_action(match_message, action_label)
+
+        # the more I write code here about reacting to ThreatExchange, the more I wish it was somewhere else entirely. Implementing
+        # here forever ties up action-related code with ThreatExchange-specific code / functionality. I prefer putting
+        # ThreatExchange-related specifics for reacting into another lambda that also subscribes to the matcher's notifications.
+        if threat_exchange_reacting_is_enabled():
+            threat_exchange_reaction_labels = get_threat_exchange_reaction_labels(action_labels)
+            if threat_exchange_reaction_labels is t.List[Label]:
+                for threat_exchange_reaction_label in threat_exchange_reaction_labels:
+                    react_to_threat_exchange(match_message, threat_exchange_reaction_label)
+
+
+def get_action_labels(match_message: MatchMessage) -> t.List[ActionLabel]:
+    """
+    TODO finish implementation
+    Returns an ActionLabel for each ActionRule that applies to a MatchMessage. 
+    """
+    action_rules = get_action_rules()
+    action_labels = [] # use Set here
+    for action_rule in action_rules:
+        if action_rule.applies(match_message) and not action_labels.contains(action_rule.action_label):
+            action_labels.append(action_rule.action_label)
+    action_labels = remove_superseded_actions(action_labels) # maybe not needed for v0
+    return action_labels
+
+
+def get_action_rules() -> t.List[ActionRule]:
+    """
+    TODO implement
+    Returns a collection of ActionRule objects. An ActionRule will have the following attributes:
+    MustHaveLabels, MustNotHaveLabels, ActionLabel
+    """
+    return [ActionRule(ActionLabel("Action", "EnqueueForReview"), [Label("Collaboration", "12345")], [])]
+
+
+def get_actions() -> t.List[Action]:
+    """
+    TODO implement
+    Returns a collection of Action objects. An Action will have the following attributes:
+    ActionLabel, Priority, SupersededByActionLabel
+    """
+    return [Action(ActionLabel("Action", "EnqueueForReview"), 1, [ActionLabel("Action", "SomeMoreImortantAction")])]
+
+
+def remove_superseded_actions(action_labels: t.List[ActionLabel]) -> t.List[ActionLabel]:
+    """
+    TODO implement
+    Evaluates a collection of ActionLabels against the configured Action objects, removing
+    an ActionLabel when it's superseded by another.
+    """
+    return action_labels
+
+
+def threat_exchange_reacting_is_enabled() -> bool:
+    """
+    TODO implement
+    Looks up whether ThreatExchange reactions are enabled.
+    """
+    return True
+
+
+def get_threat_exchange_reaction_labels(action_labels: t.List[ActionLabel]) -> t.List[Label]:
+    """
+    TODO implement
+    Evaluates a collection of action_labels against some yet to be defined configuration
+    (and possible business login) to produce 
+    """
+    return [Label("ThreatExchangeReaction", "SawThisToo")]
```

Comment:
Maybe we want to create a class `ThreatExchangeReactionLabel`?

Reply:
@jeberl Yep was thinking that, too. Was hoping for some kind of comment to push that way.